client/src/components/dashboard/UserRecipe.jsx

```diff
@@ -1,12 +1,67 @@
-import isEmpty from 'lodash/isEmpty';
+import _ from 'lodash';
 import React, { Component } from 'react';
 import PropTypes from 'proptypes';
+import ReactPaginate from 'react-paginate';
 
-import { RecipeCardSmall } from '../recipes/Index.jsx';
-import { WelcomeDisplay } from './Index.jsx';
+import { RecipeCardSmall } from '../recipes';
+import { WelcomeDisplay } from './';
+import { Loader } from '../main';
 
 class UserRecipe extends Component {
-  hasRecipes = () => !isEmpty(this.props.userRecipes.recipes)
+  componentWillReceiveProps(nextProps) {
+    const { recipes } = this.props.userRecipes;
+    const currentPageSize = Object.keys(recipes || {}).length;
+    const nextPageSize = Object.keys(
+      nextProps.userRecipes.recipes || {}
+    ).length;
+    if (currentPageSize !== nextPageSize) {
+      const currentPage = localStorage.getItem('currentPageUserRecipes');
+      const nextRecipes = this.hasRecipes() && nextProps.userRecipes.recipes;
+      if (Object.keys(nextRecipes || {}).length === 0 && currentPage > 1) {
+        localStorage.setItem('currentPageUserRecipes', currentPage - 1);
+        this.props.isFetching(true, 'UserRecipes');
+        return this.props.fetchUserRecipes(currentPage - 1);
+      }
+    }
+  }
+
+  setPagination = () => {
+    if (this.hasRecipes()) {
+      const { pagination: { page, pageCount } } = this.props.userRecipes;
+      return {
+        page,
+        pageCount
+      };
+    }
+    return {
+      page: 1,
+      pageCount: 1
+    };
+  }
+
+  hasRecipes = () => {
+    const { userRecipes } = this.props;
+    if (_.isEmpty(userRecipes)) {
+      return false;
+    }
+    if (!_.isEmpty(userRecipes) && _.has(userRecipes, 'recipes')) {
+      if (userRecipes.pagination.totalCount === 0) {
+        return false;
+      }
+      return true;
```

Comment:
consider taking `!_.isEmpty(userRecipes)` out from the if statement, since you've initially checked if the `userRecipes ` object is empty.

Also the nested if statement could be refactored like this
```
if( _.has(userRecipes, 'recipes') && userRecipes.pagination.totalCount === 0) {
 return false;
}
```

Reply:
tried this implementation but had some bugs with it, considering refactoring